### DISCUSSION TITLE
How HTTP Range Requests can get the chunks from the rauc bundle on the HTTP server?

### DISCUSSION THREAD

**Discussion by michtw** [rauc/rauc#1582] [Q&A]
I can't understand how the update of HTTP streaming + NBD works.
Since rauc bundle on HTTP server is a squashfs file which means it's a compressed file.

As I know that the range of HTTP range requests was calculated from original uncompressed image(ext4).

How HTTP range requests can get the data from the rauc bundle(squashfs) on HTTP server?

**Comment by jluebbe** [comment]
The bundle is mounted via a NBD device, so the HTTP range requests are for the ranges as appropriate for the bundle squashfs. Enrico has given a talk about this: https://archive.fosdem.org/2023/schedule/event/delta_like_ota_streaming/

**Reply by michtw** [reply]
Hi @jluebbe 

Thanks for your answer. 
 I still got some questions about `block-hash-index` adaptive update.

1. How to get chunks of data (e.g. rootfs.ext4) since the bundle is not mounted at the HTTP server? 

2. How to mount a bundle via a NBD device?  Since the bundle is not on the target device.

3. What's the data format that HTTP range requests return?  Are they the chunks of the remote bundle itself (squashfs) or chunks of image (e.g. rootfs.ext4) in the bundle?
     In short, are the chunks downloaded from  HTTP range requests compressed or not?

Thanks.
